README.md

```diff
@@ -0,0 +1,326 @@
+        type: <string|nil>
+        options: <map[string]interface{}>
+```
+
+#### Autogenerated Private Key
+
+Configures a private key source which generates key pairs automatically and publishes them to a key server.
+
+```yaml
+private_key:
+  type: autogenerated
+  options:
+    # How often we publish a new key
+    rotate_every: <time.Duration|12h>
+
+    # Folder in which to store autogenerated keys on disk
+    key_folder: <string|~/.config/jwtproxy/>
+
+    # Registerable key server and config at which to publish public keys
+    key_server:
+      type: <string|nil>
+      options: <map[string]interface{}>
+```
+
+#### Key Registry Key Server
+
+Configures a key server which talks to a server which implements the key registry protocol.
+
+```yaml
+key_server:
+  type: keyregistry
+  options:
+    # Base URL from which to access key registry endpoints.
+    registry: <string|nil>
+```
+
+#### Preshared Private Key
+
+Configures a private key source which simply uses the key files specified.
+
+```yaml
+private_key:
+  type: preshared
+  options:
+    # Unique identifier for the private key
+    key_id: <string|nil>
+
+    # Location of PEM encoded private key file
+    private_key_path: <path|nil>
+```
+
+### Verifier Config
+
+Configures and enables one or more JWT verifying reverse proxyies.
+
+```yaml
+jwtproxy:
+  verifier_proxies:
+  - enabled: <bool|true>
+
+    # Addr at which to listen for requests
+    # It can either be an HTTP(s) URL or an UNIX socket path prefixed by 'unix:'
+    listen_addr: <string|:8081>
+    shutdown_timeout: <time.Duration|1m>
+
+    # Optional PEM private key and certificate files for SSL termination
+    key_file: <path|nil>
+    crt_file: <path|nil>
+
+    verifier:
+      # Upstream server to which to forward requests
+      # It can either be an HTTP(s) URL or an UNIX socket path prefixed by 'unix:'
+      upstream: <string|nil>
+
+      # Required value for audience claim,
+      # Usually our advertised protocol and hostname
+      audience: <string|nil>
+
+      # How much time skew we allow between signer and verifier
+      max_skew: <time.Duration|1m>
+
+      # Maximum total amount of time for which a JWT can be signed
+      max_ttl: <time.Duration|5m>
+
+      # URL where unauthenticated requests should be redirected to
+      # If null then 403 response code will be responded without redirection
+      auth_redirect_url: <string|nil>
+
+      # Configures whether token should be searched in cookie `access_token`
+      # `false` by default, so JwtProxy ignores token specified in cookies
+      cookies_enabled: <bool|nil>
+
+      # Registerable key server type and options used to fetch
+      # public keys for verifying signatures
+      key_server:
+         type: <string|nil>
+         options: <map[string]interface{}>
+
+      # Registerable type and options where we track used nonces
+      nonce_storage:
+        type: <string|nil>
+        options: <map[string]interface{}>
```

Comment:
pls add excludes examle

Reply:
Thanks for reminding. I'll add

Reply:
Added here https://github.com/eclipse/che-jwtproxy/pull/8/files#diff-04c6e90faac2675aa89e2176d2eec7d8R185